### PR TITLE
[LMC-15] - Job operations #40

### PR DESCRIPTION
**Pull request by k1r1k1**
Issue #40 

List of all user jobs created:

- Jobs could be removed;
- Jobs could be paused;
- Jobs could be resumed;
- Users can see exhaustive information about all jobs;
- Users can get a list of items for each job;

### CODE REVIEW

File: src/components/house/HouseJob.js [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+} from 'react-native';
+import { withNavigation } from 'react-navigation';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import actions from '../../store/actions/index';
+import IonIcon from '../core/IonIcon';
+import { JOB_WATCH_SCREEN } from '../../constants/Routes';
+import * as Colors from '../../constants/Colors';
+
+class houseJob extends React.PureComponent {
```
Comment (Fedkasin) [inline src/components/house/HouseJob.js]: Please use upper camel case in class names `HouseJobComponent`

File: src/components/house/HouseFilter.js [hunk 2]
```diff
@@ -24,18 +25,14 @@ const styles = StyleSheet.create({
...
-  },
 });
 
 class HouseFilter extends React.PureComponent {
   render() {
     const { filters, handlers } = this.props;
     return (
       <View style={styles.container}>
-        <Text style={{ fontSize: 24, padding: 10 }}>Rooms count</Text>
+        <Text style={{ fontSize: 20, padding: 10 }}>Rooms count</Text>
```
Comment (Fedkasin) [inline src/components/house/HouseFilter.js]: Can we move this to styles?

File: src/components/house/HouseFilter.js [hunk 3]
```diff
@@ -55,7 +52,7 @@ class HouseFilter extends React.PureComponent {
           />
         </View>
         <View style={styles.divider} />
-        <Text style={{ fontSize: 24, padding: 10 }}>Price</Text>
+        <Text style={{ fontSize: 20, padding: 10 }}>Price</Text>
```
Comment (Fedkasin) [inline src/components/house/HouseFilter.js]: The same

File: src/components/house/HouseJob.js [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+  render() {
+    const {
+      index, item, onPlayPause, onClose, iosIcon, otherIcon, isEditing, navigation,
+    } = this.props;
+    const styles = StyleSheet.create({
```
Comment (Fedkasin) [inline src/components/house/HouseJob.js]: Please move this outside the class

File: src/components/house/HouseJob.js [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+              <IonIcon
+                name={Platform.OS === 'ios' ? 'ios-home' : 'md-home'}
+                color={Colors.gray}
+              />
+              <Text style={{ color: Colors.gray, fontSize: 16, fontWeight: '500' }}>
+                { item.params.rooms.toString() }
+              </Text>
+            </View>
+            <TouchableOpacity
+              onPress={() => {
```
Comment (Fedkasin) [inline src/components/house/HouseJob.js]: Don't use the anonymous functions
Create a class method

File: src/helpers/lmapi.js [hunk 6]
```diff
@@ -12,30 +12,27 @@ class LMapi {
   getCurrentUserJobs = async () => {
```
Comment (Fedkasin) [inline src/helpers/lmapi.js:12]: Please add try/catch

File: src/helpers/lmapi.js [hunk 7]
```diff
@@ -12,30 +12,27 @@ class LMapi {
...
-      return jobs;
+      return res.body;
     } else {
       return null;
     }
   };
 
   stopAllCurrentUserJobs = async () => {
-    const jobs = await this.getCurrentUserJobs();
+    const req = await this.getCurrentUserJobs();
```
Comment (Fedkasin) [inline src/helpers/lmapi.js]: The same

File: src/helpers/lmapi.js [hunk 8]
```diff
@@ -51,6 +48,30 @@ class LMapi {
     await req;
   }
 
+  removeCurrentUserJob = async (jobId) => {
```
Comment (Fedkasin) [inline src/helpers/lmapi.js:63]: The same

File: src/helpers/lmapi.js [hunk 9]
```diff
@@ -51,6 +48,30 @@ class LMapi {
...
+    // const token = await AsyncStorage.getItem('@UserStore:API_TOKEN');
+    const req = superagent.delete(`${getEnvVars.apiUrl}/v1/watch/${jobId}`);
+    req.timeout(respTime);
+    // req.set('Authorization', token);
+    await req;
+  }
+
+  pauseCurrentUserJob = async (jobId) => {
```
Comment (Fedkasin) [inline src/helpers/lmapi.js:75]: The same

File: src/helpers/lmapi.js [hunk 10]
```diff
@@ -51,6 +48,30 @@ class LMapi {
...
+    // const token = await AsyncStorage.getItem('@UserStore:API_TOKEN');
+    const req = superagent.post(`${getEnvVars.apiUrl}/v1/watch/${jobId}/pause`);
+    req.timeout(respTime);
+    // req.set('Authorization', token);
+    await req;
+  }
+
+  resumeCurrentUserJob = async (jobId) => {
```
Comment (Fedkasin) [inline src/helpers/lmapi.js:87]: the same

File: src/helpers/lmapi.js [hunk 11]
```diff
@@ -63,7 +84,7 @@ class LMapi {
 
   getHouses = async (query) => {
     // const token = await AsyncStorage.getItem('@UserStore:API_TOKEN');
-    const req = superagent.get(`${getEnvVars.apiUrl}/onliner?${query}`);
+    const req = superagent.get(`${getEnvVars.apiUrl}/v1/onliner?${query}`);
```
Comment (Fedkasin) [inline src/helpers/lmapi.js]: The same

File: src/screens/JobWatchLotsScreen.js [hunk 12]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import JobWatchLotsContainer from '../containers/JobWatchLotsContainer';
+
+class JobWatchLotsScreen extends React.Component {
```
Comment (Fedkasin) [inline src/screens/JobWatchLotsScreen.js]: Please use Pure component

File: src/store/reducers/houseWatchLotsReducers.js [hunk 13]
```diff
@@ -50,13 +56,32 @@ const houseWatchLotsReducer = (state = initialState, action) => {
...
         isWatching: true,
       };
     case CHECK_HOUSE_WATCH_STATE_FALSE:
       return {
         ...state,
         isWatching: false,
         houseWatchLots: [],
+        houseWatchJobs: [],
+      };
+    case REMOVE_HOUSE_WATCH_JOB:
```
Comment (Fedkasin) [inline src/store/reducers/houseWatchLotsReducers.js]: Please check if we could remove this

File: src/store/sagas/watchHouseLotsSaga.js [hunk 14]
```diff
@@ -2,19 +2,21 @@ import { put, takeLatest, call } from 'redux-saga/effects';
...
+  PAUSE_HOUSE_WATCH_JOB,
+  RESUME_HOUSE_WATCH_JOB,
 } from '../../constants/Actions';
 
 function* checkWatchHouseLotsState() {
   try {
     const res = yield call(LMapi.getCurrentUserJobs);
-    if (res && res.length > 0) {
-      yield put(actions.houseWatchLotsActions.watchHouseLotsTrue());
+    if (res && res.message && res.message.length > 0) {
```
Comment (Fedkasin) [inline src/store/sagas/watchHouseLotsSaga.js]: Please use destructuring here

File: src/store/sagas/watchHouseLotsSaga.js [hunk 15]
```diff
@@ -50,7 +52,7 @@ function* updateHouseWatchFilterApply(action) {
       max: parseInt(filters.priceTo, 10),
       min: parseInt(filters.priceFrom, 10),
     };
-    LMapi.startCurrentUserJob(params);
+    yield call(() => LMapi.startCurrentUserJob(params));
```
Comment (Fedkasin) [inline src/store/sagas/watchHouseLotsSaga.js]: Please try without anonymous function

File: src/components/house/HouseJob.js (hunk 4 above)
Comment (k1r1k1) [reply src/components/house/HouseJob.js]: Props should stay here

File: src/components/house/HouseJob.js (hunk 5 above)
Comment (k1r1k1) [reply src/components/house/HouseJob.js]: The absence of anonymous function as a wrapper is cause of one false triggering of an event.

File: src/components/house/HouseJob.js [hunk 16]
```diff
@@ -0,0 +1,218 @@
...
+  }
+
+  render() {
+    const {
+      index, item, onPlayPause, onClose, iosIcon, otherIcon, isEditing, navigation, onUpdateHouseWatchLots,
+    } = this.props;
+    return (
+      <View style={styles.container}>
+        <View style={styles.job}>
+          <View style={{
```
Comment (Fedkasin) [inline src/components/house/HouseJob.js]: Please move this to styles

File: src/containers/HouseWatchLotsContainer.js [hunk 17]
```diff
@@ -1,54 +1,130 @@
...
+    paddingBottom: 10,
+    paddingLeft: 5,
+  },
+  divider: {
+    borderBottomColor: Colors.lightGray,
+    borderBottomWidth: 1,
+  },
+});
+
 class HouseWatchLotsContainer extends React.Component {
```
Comment (Fedkasin) [inline src/containers/HouseWatchLotsContainer.js]: Please use PureComponent

File: src/containers/HouseWatchLotsContainer.js [hunk 18]
```diff
@@ -1,54 +1,130 @@
...
-      <View>
+      <View style={{ flexDirection: 'column' }}>
         <View style={{ display: 'flex', flexDirection: 'row', padding: 10 }}>
           <Text style={{ fontSize: 24, color: Colors.gray, marginLeft: 9 }}>Live tracking</Text>
           <Switch
             value={isWatching}
             disabled={!isWatching}
             style={{ marginLeft: 'auto' }}
-            onValueChange={onUpdateHouseWatchState}
+            onValueChange={() => this.onStopAllJobs()}
```
Comment (Fedkasin) [inline src/containers/HouseWatchLotsContainer.js:109]: Can we just write the `onValueChange={this.onStopAllJobs}` ?

File: src/helpers/lmapi.js [hunk 19]
```diff
@@ -10,93 +10,159 @@ class LMapi {
...
+      const userId = await AsyncStorage.getItem('@UserStore:USER_ID');
+      // const token = await AsyncStorage.getItem('@UserStore:API_TOKEN');
+      const req = superagent.get(`${getEnvVars.apiUrl}/v1/users/${userId}/jobs`);
+      req.timeout(respTime);
+      // req.set('Authorization', token);
+      const res = await req;
+      const { body } = res || { body: { message: [] } };
+      if (body.message.length !== 0) {
+        return res.body;
+      } else {
```
Comment (Fedkasin) [inline src/helpers/lmapi.js]: Please remove else

File: src/helpers/lmapi.js [hunk 20]
```diff
@@ -10,93 +10,159 @@ class LMapi {
...
-    } else {
-      return null;
+      const req = superagent.get(`${getEnvVars.apiUrl}/v1/onliner?${query}`);
+      req.timeout(respTime);
+      // req.set('Authorization', token);
+      const res = await req;
+      const { body } = res || { body: { status: [] } };
+      if (body.status === 'success') {
+        return res.body;
+      } else {
```
Comment (Fedkasin) [inline src/helpers/lmapi.js]: The same

File: src/helpers/lmapi.js [hunk 21]
```diff
@@ -10,93 +10,159 @@ class LMapi {
...
-    } else {
-      return null;
+      const req = superagent.get(`${getEnvVars.apiUrl}/watch/${jobId}`);
+      req.timeout(respTime);
+      // req.set('Authorization', token);
+      const res = await req;
+      const { body } = res || { body: { status: [] } };
+      if (body.status === 'success') {
+        return res.body;
+      } else {
```
Comment (Fedkasin) [inline src/helpers/lmapi.js]: Same

File: src/helpers/lmapi.js [hunk 22]
```diff
@@ -10,93 +10,159 @@ class LMapi {
...
-    } else {
-      return null;
+      const req = superagent.get(`https://jsonplaceholder.typicode.com/photos?_start=${start}&_limit=${limit}`);
+      req.timeout(respTime);
+      // req.set('Authorization', token);
+      const res = await req;
+      const { status } = res || null;
+      if (status === 200) {
+        return res.body;
+      } else {
```
Comment (Fedkasin) [inline src/helpers/lmapi.js]: Same

**Review by Fedkasin — APPROVED** [review]

**Comment by Fedkasin** [comment]
LGTM

**Pull request merged**